pkg/controller/servicebindingrequest/servicecontext_test.go

```diff
@@ -0,0 +1,162 @@
+package servicebindingrequest
+
+import (
+	"testing"
+
+	routev1 "github.com/openshift/api/route/v1"
+	pgv1alpha1 "github.com/operator-backing-service-samples/postgresql-operator/pkg/apis/postgresql/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/pkg/testutils"
+	"github.com/redhat-developer/service-binding-operator/test/mocks"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestBuildServiceContexts(t *testing.T) {
+	ns := "planner"
+	name := "service-binding-request"
+	resourceRef := "db-testing"
+	matchLabels := map[string]string{
+		"connects-to": "database",
+		"environment": "planner",
+	}
+	f := mocks.NewFake(t, ns)
+	sbr := f.AddMockedServiceBindingRequest(name, nil, resourceRef, "", deploymentsGVR, matchLabels)
+	sbr.Spec.BackingServiceSelectors = &[]v1alpha1.BackingServiceSelector{
+		*sbr.Spec.BackingServiceSelector,
+	}
+	f.AddMockedUnstructuredCSV("cluster-service-version")
+	f.AddMockedDatabaseCR(resourceRef, ns)
+	f.AddMockedUnstructuredDatabaseCRD()
+	f.AddMockedUnstructuredSecret("db-credentials")
+
+	restMapper := testutils.BuildTestRESTMapper()
+
+	t.Run("existing selectors", func(t *testing.T) {
```

Comment:
please improve test names to reflect what is the context and expectation.

Reply:
Review test names #472